tests/jsrun.py

```diff
@@ -115,7 +115,11 @@ def run_js(filename, engine=None, args=[],
   out = ['' if o is None else o for o in (stdout, stderr)]
   ret = '\n'.join(out) if full_output else out[0]
 
-  if assert_returncode is not None and proc.returncode is not assert_returncode:
+  # assert_returncode = None means we expect a failure.
+  if assert_returncode is None:
+    if proc.returncode == 0:
```

Comment:
this surprises me. I'd expect `assert_returncode == None` to mean "don't check it". It seems to instead mean "assert it is not 0"?

Reply:
Yeah I agree its confusing... but.  Previously it meant: 

  "we don't care it fails.. i.e. don't check"

Now it means:

 "I expect this to fail, but I don't care about the exact return code"

I experimented with more intrusive changes here but could not find something I liked.    The best alternative I could think of was to have an additional argument call `expect_file=True` which would override the assert_returncode thing.. but this would need to be added in a lot of places.     Perhaps we could have a magic value called ANY_FAIL=xxxx and pass that around?

Reply:
Yeah, a special "fail with any code" value might work, `NON_ZERO` maybe could be the name? then `if assert_returncode is NON_ZERO` would make sense.

Reply:
Good idea.  Done.